base/applications/mspaint/scrollbox.cpp

```diff
@@ -176,6 +176,13 @@ LRESULT CScrollboxWindow::OnVScroll(UINT nMsg, WPARAM wParam, LPARAM lParam, BOO
 LRESULT CScrollboxWindow::OnLButtonDown(UINT nMsg, WPARAM wParam, LPARAM lParam, BOOL& bHandled)
 {
     selectionWindow.ShowWindow(SW_HIDE);
+
+    if (toolsModel.GetActiveTool() == TOOL_SHAPE)
+    {
+        toolsModel.OnButtonDown(TRUE, -1, -1, TRUE); // Finish the shape
+        imageArea.Invalidate();
+    }
```

Comment:
What bug does that fix?

Reply:
If clicking outside when `TOOL_SHAPE`, it completes the shape.

Reply:
No idea how you can click outside 🤔 The only way I find is that one can only do a mouse up to finish a shape. So if you need to do a mouse down, how can you manage that? And in that case, what happens if you do mouse-down inside the image?

Reply:
1. Select `TOOL_SHAPE`.
2. Make a partial shape, 
3. Click the gray area out of the image area.
4. The shape completes.

Reply:
On Windows, when you click on the gray area (outside the image), the shape is cancelled.

Reply:
Windows 10 has different behaviour.